src/jdk.compiler/share/classes/com/sun/tools/javac/code/Type.java

```diff
@@ -1123,7 +1115,9 @@ public <R,S> R accept(Type.Visitor<R,S> v, S s) {
         }
 
         public Type constType(Object constValue) {
-            return addMetadata(new ConstantValue(constValue));
+            return isPrimitive() ?
+                    addMetadata(new ConstantValue(constValue)) :
+                    addMetadata(new ConstantValue(constValue)).addMetadata(new TypeMetadata.NullMarker(NullMarker.NOT_NULL));
```

Comment:
One nitpick: I can see why having the call here is handy. On the other hand, in terms of code maintenance it would probably be better if `constType` added just one kind of metadata (and we added the NOT_NULL explicitly where we need it). That said, I understand that, pragmatically, Attr is probably not the only place where we call this method, and chasing all the usages might be a bit hard. So let's just keep a mental note of this, going forward.

Reply:
I see your point but yes I agree that for now it is more practical to have the call here. Thanks for all the comments!